### DISCUSSION TITLE
Arduino IDE does not recognize "__time_critical_func" attribute

### DISCUSSION THREAD

**Discussion by XFer012** [earlephilhower/arduino-pico#2362] [Q&A] [answered]
RESOLVED:
__time_critical_func is not an attribute. The correct declaration is:

`void __time_critical_func (myISRroutine)()`

and not 

`void __time_critical_func myISRroutine()`

Sorry for the useless post.
I leave it here for reference.

Hello,
Arduino IDE 2.3.2 on Windows 11, latest Arduino-Pico core v4.0.1

I'm trying to put a ISR in RAM, so I defined it as

```
#include <pico/stdlib.h>

void __time_critical_func myISRroutine()
{
...
}
```

Unfortunately the compiler does not recognize the attribute:

```
error: expected initializer before 'myISRroutine'
  350 | void __time_critical_func myISRroutine()
```

Maybe I included the wrong header? Is it not pico/stdlib.h?

Thanks!

**Comment by earlephilhower** [comment, marked as answer]
Use `__not_in_flash_func(x)` instead.
